pkg/kv/kvclient/kvcoord/dist_sender.go

```diff
@@ -452,7 +461,38 @@ type DistSenderRangeFeedMetrics struct {
 	Errors                 rangeFeedErrorCounters
 }
 
-func MakeDistSenderMetrics() DistSenderMetrics {
+// DistSenderKVInterceptorMetrics is a set of KV interceptor specific metrics.
+type DistSenderKVInterceptorMetrics struct {
+	EstimatedReplicationBytes *aggmetric.AggCounter
+
+	mu struct {
+		syncutil.Mutex
+		// cachedPathMetrics stores a cache of network paths to the metrics
+		// which have been initialized. Having this layer of caching prevents us
+		// from needing to compute the normalized locality on every request.
+		cachedPathMetrics map[networkPath]networkPathMetrics
```

Comment:
This is the rare appropriate time to use `sync.Map`. Currently `updateEstimatedReplicationBytes` is grabbing the mutex once for each replica. But using a sync map would allow avoiding mutex acquisition if the entry is already present.